src/controllers/genres.controllers.ts

```diff
@@ -0,0 +1,107 @@
+import { NextFunction, Request, Response } from 'express';
+import Genre from '../models/genre.model';
+import IGenre from '../interfaces/genre.interface';
+import CustomError from '../utils/CustomError';
+
+export const gellAllGenres = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	try {
+		const genres = await Genre.find();
+		return res.status(200).json(genres);
+	} catch (error) {
+		next(error);
+	}
+};
+
+export const createGenre = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	const { name } = req.body as IGenre;
+
+	try {
+		const createdGenre = new Genre({ name });
+		const result = await createdGenre.save();
+
+		return res.status(201).json(result);
+	} catch (error) {
+		next(error);
+	}
+};
+
+export const getGenre = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	const genreId = req.params.genreId;
+	try {
+		const genre = await Genre.findById(genreId);
+
+		if (!genre) {
+			return next(
```

Comment:
You call next() callback inside catch() block. Here you should throw SomeError and inside the catch block return next(err);

Reply:
rewrote, using `throw` error in try block and returning `next(err)` in catch